frontend/src/main/components/Ride/RideForm.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react'
+import { Button, Form } from 'react-bootstrap';
+import { useForm } from 'react-hook-form'
+import { useNavigate } from 'react-router-dom';
+function RideForm({ initialContents, submitAction, buttonLabel = "Create" }) {
+
+    
+    // Stryker disable all
+    const {
+        register,
+        formState: { errors },
+        handleSubmit,
+    } = useForm(
+        { defaultValues: initialContents || {}, }
+    );
+    // Stryker enable all
+    const navigate = useNavigate();
+
+    //const testIdPrefix = "RideForm";
+
+    return (
+
+        <Form onSubmit={handleSubmit(submitAction)}>
+
+            {initialContents && (
+                <Form.Group className="mb-3" >
+                    <Form.Label htmlFor="id">Id</Form.Label>
+                    <Form.Control
+                        data-testid={"RideForm-id"}
+                        id="id"
+                        type="text"
+                        {...register("id")}
+                        value={initialContents.id}
+                        disabled
+                    />
+                </Form.Group>
+            )}
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="day">Day</Form.Label>
+                <Form.Control
+                    data-testid={"RideForm-day"}
+                    id="day"
+                    type="text"
+                    isInvalid={Boolean(errors.day)}
+                    {...register("day", {
+                        required: "Day is required.",
+
+                    })}
+                    placeholder="eg: Monday"
+                />
+                <Form.Control.Feedback type="invalid">
+                    {errors.day?.message}
+                </Form.Control.Feedback>
+            </Form.Group>
+
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="startTime">Start Time</Form.Label>
+                <Form.Control
+                    data-testid={"RideForm-startTime"}
+                    id="startTime"
+                    type="text"
+                    isInvalid={Boolean(errors.startTime)}
+                    {...register("startTime", {
+                        required: "Start time is required.",
+
```

Comment:
Validate this input with a regular expression of some kind.  This is a best guess, not guaranteed to be correct:

```suggestion
 pattern: /(01|02|03|04|05|06|07|08|09|10|11|12):[0-5][0-9][AP]M/,
```

https://www.react-hook-form.com/api/useform/register